sunpy/timeseries/timeseriesbase.py

```diff
@@ -334,6 +335,10 @@ def concatenate(self, otherts, **kwargs):
         at once.
         """
 
+        # check to see if nothing needs to be done
+        if self == otherts:
```

Comment:
It checks that the data, units and metadata for the two TS match. If so they are considered to be equivalent.
In this case I used Pandas functionality to compare the data DataFrames and compared the list storing metadata (tuple) entries in the metadata. The units are simply a dictionary and this is compared directly.